Restructure igvc-electrical repo

Cryptum169 opened the pull request:
* Moved motorboard into discont
* Added a `README.md` explaining difference of motorboard and logic board.
* Placed all hotswap board files into hotswap
* Grouped diagnostic, estop, hotswap, and logic into a single core-boards folder. Since we have too many boards to put on root directory like roboracing-electrical.
* Placed overall folder into docs since it's supposed to show the overall schematics of the system.

Also don't worry about mbed-firmware. It is /motorboard/mbed-firmware instead of logic/mbed-firmware so we good.

Closes #36.

Commit:
restructure

Cryptum169 commented:
#36

oswinso commented:
You should probably give a small description of the changes you made so that it's easy for other people without context to understand the changes you're making.

It looks like you deleted some files and then removed the mbed-firmware folder?

Cryptum169 commented:
My bad, what I did are in the following:

Moved `motorboard` into `discont`, added a `readme.md` explaining difference of motorboard and logic board.
Placed all hotswap board files into `hotswap`
Grouped `diagnostic`, `estop`, `hotswap`, and `logic` into a single `core-boards` folder. Since we have too many boards to put on root directory like `roboracing-electrical`.
Placed `overall` folder into `docs` since it's supposed to show the overall schematics of the system.
Also don't worry about `mbed-firmware`. It is `/motorboard/mbed-firmware` instead of `logic/mbed-firmware` so we good.

oswinso commented:
I edited the PR description to include what you just wrote and formatted it a bit.

Cryptum169 commented:
Approved per igvc-electrical meeting on July 28th, 2019.
